### PR TITLE
Make sure the working branch can be deleted

### LINKED ISSUE #284: The working branches are not deleted of labsync
Let's make things better!

Labsync checkout a working branch every working day, and try to delete the branch. But there is a logic error in the code, it checks whether there is something modified, if there is not, the 'set -e' make the script returned, the result is the working branch is not deleted, the code is bellow:

set -e
......
is_modified="`hub status |grep modified`"
if [[ $is_modified ]];then
    ...... //summit pr here
fi

hub checkout master
hub branch -D ${branch_name} //delete branch here

### PR DESCRIPTION
**Pull request by huangtianhua**
Make sure the working branches can be deleted for labsync.

Closes: theopenlab/openlab#284

### CODE REVIEW

File: labsync/sync_prepare.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ hub checkout master
 
 # maybe some errors happened last time, the branch wont be clean
 is_clean="`hub status |grep clean`"
```
Comment (bzhaoopenstack) [inline labsync/sync_prepare.sh]: so is_clean is not used anymore.

Comment (kiwik) [reply labsync/sync_prepare.sh]: Good catch, `is_modified` is same issue.

Comment (huangtianhua) [reply labsync/sync_prepare.sh]: Sorry, forgot delete it.

**Comment by huangtianhua** [comment]
recheck

**Comment by huangtianhua** [comment]
recheck

**Comment by Yikun** [comment]
recheck

**Comment by wangxiyuan** [comment]
recheck

**Comment by wangxiyuan** [comment]
recheck

**Comment by huangtianhua** [comment]
recheck

**Comment by huangtianhua** [comment]
recheck

**Comment by huangtianhua** [comment]
recheck

**Review by wangxiyuan — APPROVED** [review]

**Pull request merged**